chess-AI/train.py

```diff
@@ -22,7 +22,7 @@ class Train:
         mcts: References the MCTS class
         policy_converter: References the PlayNetworkPolicyConverter class
     """
-    def __init__(self, lr, move_approximator, val_approximator=None):
+    def __init__(self, lr, move_approximator, save_path, val_approximator=None):
```

Comment:
Since specifying a save path is optional, let's have this parameter default to `None` e.g. `save_path=None`.